requests.go

```diff
@@ -4,13 +4,20 @@ import (
 	"context"
 	"net/http"
 	"path"
+	"strings"
 	"sync"
 	"time"
 
 	"github.com/pkg/errors"
 	"golang.org/x/sync/errgroup"
 )
 
+var NotSupportRequestRange = errors.New("does not support range request")
```

Comment:
The error variable should have `Err` prefix in Go
```suggestion
var ErrNotSupportRequestRange = errors.New("does not support range request")
```

Reply:
This also won't work, because the error will be wrapped into another error

Reply:
Works fine
https://go.dev/play/p/YbEfSE_rm9g

Reply:
github.com/pkg/errors   v0.9.1 works fine